### PR TITLE
lightning: support keep checkpoint data after success without changing schema name

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?
DM depending load task checkpoint data to judge whether the load phase is finished. Since DM can auto create a unique checkpoint schema name for each load task, we shouldn't change the checkpoint schema when use lightning to replace loader

### What is changed and how it works?
Extend the `keep-after-success` config field from boolean value  with three enum:
- `remove`. Same as `false`, remove checkpoint file/schema after success
- `rename`. Same as `true`, keep checkpoint data, but auto rename with a suffix `.{taskID}.bak`
- `origin`. New config, keep the original checkpoint data by do nothing.

### Check List 

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
Support keep checkpoint data after success without changing schema name
```

### CODE REVIEW

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Comment by lance6716** [comment]
(needed by https://github.com/pingcap/dm/pull/2051 )

File: br/pkg/lightning/config/config.go [hunk 1]
```diff
@@ -242,6 +242,74 @@ func (t PostOpLevel) String() string {
...
+	case bool:
+		if val {
+			*t = CheckpointRename
+		} else {
+			*t = CheckpointRemove
+		}
+	case string:
+		return t.FromStringValue(val)
+	default:
+		return errors.Errorf("invalid op level '%v', please choose valid option between ['off', 'optional', 'required']", v)
```
Comment (lance6716) [inline br/pkg/lightning/config/config.go:267]: remove, rename, origin, and below default branch

**Review by lance6716 — COMMENTED** [review]
rest lgtm

**Review by lance6716 — APPROVED** [review]

**Pull request merged**